powerscale/provider/storagepool_tier_datasource.go

```diff
@@ -0,0 +1,280 @@
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/helper"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &StoragepoolTierDataSource{}
+
+// NewStoragepoolTierDataSource creates a new data source.
+func NewStoragepoolTierDataSource() datasource.DataSource {
+	return &StoragepoolTierDataSource{}
+}
+
+// StoragepoolTierDataSource defines the data source implementation.
+type StoragepoolTierDataSource struct {
+	client *client.Client
+}
+
+// Metadata describes the data source arguments.
+func (d StoragepoolTierDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_storagepool_tier"
+}
+
+// Schema describes the data source arguments.
+func (d StoragepoolTierDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "This datasource is used to query the Storagepool tiers from PowerScale array. " +
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Description: "This datasource is used to query the Storagepool tiers from PowerScale array. " +
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				MarkdownDescription: "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				Optional:            false,
```

Comment:
No need to mention attributes if its false

Reply:
removed